cairo/src/utils/dict.cairo

```diff
@@ -136,3 +137,83 @@ func hashdict_write{poseidon_ptr: PoseidonBuiltin*, dict_ptr: DictAccess*}(
     let dict_ptr = dict_ptr + DictAccess.SIZE;
     return ();
 }
+
+func hashdict_finalize{range_check_ptr}(
+    dict_ptr_start: DictAccess*,
+    dict_ptr: DictAccess*,
+    original_mapping_start: DictAccess*,
+    original_mapping_end: DictAccess*,
+    should_merge: felt,
+) -> (DictAccess*, DictAccess*) {
+    alloc_locals;
+
+    if (should_merge == FALSE) {
+        let (squashed_dict_start: DictAccess*) = alloc();
+        let (squashed_dict_end) = squash_dict(dict_ptr_start, dict_ptr, squashed_dict_start);
+        let (prev_values_start, prev_values_end) = prev_values(
+            squashed_dict_start, squashed_dict_end
+        );
+        tempvar dict_ptr_start = prev_values_start;
+        tempvar dict_ptr = prev_values_end;
+        tempvar range_check_ptr = range_check_ptr;
+    } else {
+        tempvar dict_ptr_start = dict_ptr_start;
+        tempvar dict_ptr = dict_ptr;
+        tempvar range_check_ptr = range_check_ptr;
```

Comment:
reorganize the range_check and args to avoid doing useless `tempvar` when calling `prev_values`

```suggestion
    } else {
        tempvar range_check_ptr = range_check_ptr;
        tempvar dict_ptr_start = dict_ptr_start;
        tempvar dict_ptr = dict_ptr;
```